elfi/methods/mcmc.py

```diff
@@ -269,8 +270,16 @@ def _build_tree_nuts(params, momentum, log_slicevar, step, depth, log_joint0,
         log_joint = target(params1) - 0.5 * momentum1.dot(momentum1)
         n_ok = float(log_slicevar <= log_joint)
         sub_ok = log_slicevar < (1000. + log_joint)  # check for diverging error
+        if not sub_ok:
+            if np.isinf(target(params1)):  # logpdf(params1) = -inf i.e. pdf(params1) = 0 i.e. not allowed
```

Comment:
Hmm... Yes, that could be a nice clarification for potentially poor acceptance ratio.